lib/datadog/di/logging.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Datadog
+  module DI
+    # Logging for DI
+    module Logging
+      # Logs an internal warning message.
+      # When verbose logging is enabled, the logging happens at warn level.
+      # When verbose logging is disabled, the logging happens at debug level
+      # (which is how the rest of the library is reporting its internal
+      # warnings/errors).
+      def log_warn_internal(msg)
+        if settings.dynamic_instrumentation.internal.verbose_logging
+          logger.warn(msg)
+        else
+          logger.debug(msg)
+        end
+      end
+      
+      # Logs an internal informational message.
+      # When verbose logging is enabled, the logging happens at info level.
+      # When verbose logging is disabled, nothing is logged.
+      def log_info_internal(msg)
+        if settings.dynamic_instrumentation.internal.verbose_logging
+          logger.info(msg)
+        end
+      end
```

Comment:
Minor: This is a bit unexpected to me; why not log it at debug level as well? E.g. it seems weird that if I enable debug logging with `DD_TRACE_DEBUG=true`, there's still some logging that won't show up?